### DISCUSSION TITLE
Setup proxy values for artifact repository

### DISCUSSION THREAD

**Discussion by anetbnd** [argoproj/argo-workflows#12358] [Q&A] [answered]
Hello

We are currently doing our first steps with argo workflow. Our cluster is behind a proxy. The url of the proxy is dynamically generated and is changing from time to time. Every time the url is stored in a secret on the cluster, so it can be consumed by deployments or other configurations.

We can set the proxy url (http_proxy/https_proxy) from this secret without issues for a workflow template, by using the approach (https://argoproj.github.io/argo-workflows/walk-through/secrets/) described in the walk-through. However, when using S3 as repository for Artifacts we still get a timeout from the wait container sidecar. We noticed, that the wait container sidecar does not have the environment variables, defined for the workflow template. They are only set for the main container.

Is there any way to set the environment variables for the wait sidecar container, so that we don't get timeouts when using S3 as artifact repository? We could not find a clear answer in the documentation.

Thanks for the help,
André

**Comment by agilgur5** [comment, marked as answer]
The `wait` container is an [executor container](https://github.com/argoproj/argo-workflows/blob/a15587755e250fbbe3b9538b1c15ae581a108f0c/workflow/controller/workflowpod.go#L605) (along with the `init` container). Within a template, there are currently only a few explicit options to [configure the `executor`](https://argo-workflows.readthedocs.io/en/latest/fields/#executorconfig).

Within the `workflow-controller-configmap`, [more options are available](https://github.com/argoproj/argo-workflows/blob/a15587755e250fbbe3b9538b1c15ae581a108f0c/docs/workflow-controller-configmap.yaml#L235), including `env`.

You might also be able to do this at a template-level using `podSpecPatch` to override the `env`.

**Reply by anetbnd** [reply]
Thanks a lot, I tried it out with the configmap and it worked.
